### PR TITLE
Add unix_date Spark function

### PR DESCRIPTION
**Pull request by acvictor**
Returns the number of days since 1970-01-01.

Addresses #8720

### CODE REVIEW

**Comment by acvictor** [comment]
@PHILO-HE @rui-mo please review, thank you!

**Comment by PHILO-HE** [comment]
This pr looks good to me. Maybe, better to use "unix_date" in the title to align Spark's function name.
@mbasmanova, do you have any comment?

**Comment by acvictor** [comment]
@mbasmanova can you please take a look at this PR?

File: velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp [hunk 1]
```diff
@@ -114,6 +114,23 @@ TEST_F(DateTimeFunctionsTest, weekOfYear) {
   EXPECT_EQ(15, weekOfYear("2013-04-08"));
 }
 
+TEST_F(DateTimeFunctionsTest, unixDate) {
+  const auto unixDate = [&](std::optional<int32_t> value) {
+    return evaluateOnce<int32_t, int32_t>("unix_date(c0)", {value}, {DATE()});
+  };
+
+  EXPECT_EQ(unixDate(util::fromDateString("1970-01-01")), 0);
```
Comment (mbasmanova) [inline velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp:122]: There is a lot of repetition here. Consider changing unixDate lambda to take date string as input and re-write this code as 

```
EXPECT_EQ(unixDate(1970-01-02"), 1);
```

File: velox/docs/functions/spark/datetime.rst [hunk 2]
```diff
@@ -163,6 +163,14 @@ These functions support TIMESTAMP and DATE input types.
 
     Alias for ``unix_timestamp(string, format) -> integer``.
 
+.. spark:function:: unix_date(date) -> integer
+
+    Returns the number of days since 1970-01-01.
```
Comment (mbasmanova) [inline velox/docs/functions/spark/datetime.rst:168]: Would you generate the docs and check formatting? It looks like you might be missing :: before the examples.

File: velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp [hunk 3]
```diff
@@ -114,6 +114,24 @@ TEST_F(DateTimeFunctionsTest, weekOfYear) {
   EXPECT_EQ(15, weekOfYear("2013-04-08"));
 }
 
+TEST_F(DateTimeFunctionsTest, unixDate) {
+  const auto unixDate = [&](const StringView& date) {
```
Comment (mbasmanova) [inline velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp:118]: const StringView& -> std::string_view

use std::string_view and pass it by value

File: velox/docs/functions/spark/datetime.rst (hunk 2 above)
Comment (acvictor) [reply velox/docs/functions/spark/datetime.rst]: Done, thanks!

File: velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp (hunk 1 above)
Comment (acvictor) [reply velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp]: Updated.

File: velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp (hunk 3 above)
Comment (acvictor) [reply velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp]: Updated.

**Comment by acvictor** [comment]
@mbasmanova can you review again please? Thanks!

**Review by mbasmanova — APPROVED** [review]
Looks good to me.

**Review by Yuhta — APPROVED** [review]

**Comment by acvictor** [comment]
> Looks good to me.

Thank you for the review!

**Pull request closed without merging**